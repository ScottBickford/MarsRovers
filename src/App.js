import React from "react";
import "./App.css";
import { Container } from "semantic-ui-react";
import RoverInstructionsPage from "./features/roverInstructions/RoverInstructionsPage";

function App() {
  return (
    <Container className='main'>
      <h2>Rover Instructions</h2>
      <RoverInstructionsPage />
    </Container>
  );
}

export default App;
