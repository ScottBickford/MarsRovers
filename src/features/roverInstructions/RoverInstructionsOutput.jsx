import React from 'react';
import { Segment, Header } from 'semantic-ui-react';
import RoverInstructionsOutputItem from './RoverInstructionsOutputItem';

export const RoverInstructionsOutput = ({ roverMovements }) => {
  const hasErrors = roverMovements.rovers.some(
    r => r.positions[r.positions.length - 1].error !== null
  );
  return (
    <Segment.Group>
      <Segment inverted color={hasErrors ? 'red' : 'green'}>
        <Header>Output</Header>
      </Segment>
      <Segment attached>
        {roverMovements.rovers.map(rover => {
          return <RoverInstructionsOutputItem key={rover.id} rover={rover} />;
        })}
      </Segment>
    </Segment.Group>
  );
};
