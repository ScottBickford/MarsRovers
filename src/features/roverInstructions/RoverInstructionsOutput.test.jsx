import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { RoverInstructionsOutput } from './RoverInstructionsOutput';
import { roverMovementsTestObject } from '../../App.test';

describe('RoverInstructionsOutput', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<RoverInstructionsOutput roverMovements={roverMovementsTestObject} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('passes all props to Output', () => {
    const { getByText } = render(<RoverInstructionsOutput roverMovements={roverMovementsTestObject} />);
    const buttonElement = getByText(/Output/i);
    expect(buttonElement).toBeInTheDocument();
  });
});