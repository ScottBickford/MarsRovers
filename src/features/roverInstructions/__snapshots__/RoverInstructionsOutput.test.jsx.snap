// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoverInstructionsInput snapshot renders 1`] = `
<div
  className="ui segments"
>
  <div
    className="ui green inverted segment"
  >
    <div
      className="ui header"
    >
      Output
    </div>
  </div>
  <div
    className="ui attached segment"
  >
    <div>
      1
       
      3
       
      N
    </div>
    <div>
      5
       
      1
       
      E
    </div>
  </div>
</div>
`;

exports[`RoverInstructionsOutput snapshot renders 1`] = `
<div
  className="ui segments"
>
  <div
    className="ui green inverted segment"
  >
    <div
      className="ui header"
    >
      Output
    </div>
  </div>
  <div
    className="ui attached segment"
  >
    <div>
      1
       
      3
       
      N
    </div>
    <div>
      5
       
      1
       
      E
    </div>
  </div>
</div>
`;
